src/Simulation/QsharpCore/Targeting.qs

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+/// # Summary
+/// This namespace includes Q# core functions and operations.
+namespace Microsoft.Quantum.Targeting {
+    /// # Summary
+    /// Compiler-recognized attribute used to mark a callable with the runtime capabilities it
+    /// requires.
+    ///
+    /// # Named Items
+    /// ## Level
+    /// The name of the runtime capability level required by the callable.
+    ///
+    /// ## Reason
+    /// A description of why the callable requires this runtime capability.
+    /// 
+    /// # Remarks
+    /// The valid capability level names, in order of increasing capabilities (or decreasing
+    /// restrictions), are:
+    ///
+    /// ## BasicQuantumFunctionality
+    /// Measurement results cannot be compared for equality.
+    ///
+    /// ## BasicMeasurementFeedback
+    /// Measurement results can be compared for equality only in if-statement conditional
+    /// expressions in operations. The block of an if-statement that depends on a result cannot
+    /// contain set statements for mutable variables declared outside the block, or return
+    /// statements.
+    ///
+    /// ## FullComputation
+    /// No runtime restrictions. Any Q# program can be executed.
+    @Attribute()
```

Comment:
An `/// # Example` section may help here.
```suggestion
    /// # Example
    /// The following snippet defines an operation `Reset` that requires the `BasicQuantumFunctionality` capability:
    /// ```Q#
    /// @RequiresCapability(
    ///     "BasicQuantumFunctionality",
    ///     "Reset is replaced by a supported implementation on all execution targets."
    /// )
    /// operation Reset(target : Qubit) : Unit {
    ///     // ...
    /// }
    /// ```
    @Attribute()
```

Reply:
I'm not opposed to an example, but I'm not sure we should use an example that is only a short-term workaround until the decompositions for `Reset` are integrated into the runtime. But I can't think of an actual example that seems like good practice and a recommended usage - the compiler really should infer it correctly... @bettinaheim?

Reply:
Ah, good point. Maybe something in the remarks explaining that users shouldn't use directly, then?

Reply:
I added a note about that.